src/app/loadout-builder/LoadoutBuilderContainer.tsx

```diff
@@ -5,15 +5,22 @@ import { useLoadStores } from 'app/inventory/store/hooks';
 import { Loadout } from 'app/loadout-drawer/loadout-types';
 import { useD2Definitions } from 'app/manifest/selectors';
 import { setSearchQuery } from 'app/shell/actions';
+import ErrorPanel from 'app/shell/ErrorPanel';
 import { useThunkDispatch } from 'app/store/thunk-dispatch';
 import React, { useEffect } from 'react';
 import { useSelector } from 'react-redux';
 import { useLocation } from 'react-router';
+import { createSelector } from 'reselect';
 import { DestinyAccount } from '../accounts/destiny-account';
 import { savedLoadoutParametersSelector } from '../dim-api/selectors';
-import { sortedStoresSelector } from '../inventory/selectors';
+import { allItemsSelector, sortedStoresSelector } from '../inventory/selectors';
 import LoadoutBuilder from './LoadoutBuilder';
 
+const disabledDueToMaintenanceSelector = createSelector(
+  allItemsSelector,
+  (items) => items.length > 0 && items.every((item) => item.missingSockets)
+);
```

Comment:
I'm suspicious of this logic -- `missingSockets` indicates "whether the item *should* have socket info but doesn't"
https://github.com/DestinyItemManager/DIM/blob/3779128751df640f8329ef439675e6b87ab638be/src/app/inventory/store/sockets.ts#L73-L83

so as long as the user has a single item in their inventory where the def has no sockets this will never be true.

Reply:
Fair callout - @ryan-rushton what do you think this logic should be?

Reply:
Should probably be something like `items.length > 0 && items.every((item) => item.missingSockets || !item.sockets)` (item.sockets implies item.missingSockets, i.e. every item that has sockets had its sockets built from definitions due to missing data).